### PR TITLE
Fix timeout retry and don't fail with error if ignoring vulnerabilties

### PR DESCRIPTION
**Pull request by ericvn**
Fixes: https://github.com/istio/istio/issues/32096

Fixed the current code to retry on timeouts. Also, if ignoreVulnerabilities is set, we should not fail on any scanner errors. Just log and continue.

### CODE REVIEW

File: pkg/build/scanner.go [hunk 1]
```diff
@@ -63,8 +63,9 @@ func Scanner(manifest model.Manifest, githubToken string) error {
 		if err != nil {
 			if err, ok := err.(net.Error); ok && err.Timeout() {
 				fmt.Println("Scanner request timed out. Need to run request again.")
+			} else {
+				break
```
Comment (howardjohn) [inline pkg/build/scanner.go:67]: we never return the error now as far as I can tell

Comment (ericvn) [reply pkg/build/scanner.go:67]: err should be set and returned in the check immediately after the `for` loop

Comment (ericvn) [reply pkg/build/scanner.go:67]: That said, I can simply return the err here with the same behavior.

**Review by brian-avery — APPROVED** [review]

**Pull request merged**